SPIFlashLogger.device.lib.nut

```diff
@@ -0,0 +1,847 @@
+        } else
+            _pos_write = pos;
+    }
+};
+
+class SPIFlashLogger.LoggerSerializedObject {
+    _flash = null;
+    _logger = null;
+    _addr = null;
+    _isValid = false;
+    _payload = null;
+    _len = 0;
+
+
+_debug = false;
+    constructor(addr, logger, flash) {
+        _logger = logger;
+        _flash = flash;
+        _addr = addr;
+
+        _isValid = false;
+        _payload = null;
+        _len = 0;
+
+        init(addr);
+    }
+
+    function init(pos) {
+        local requested_pos = pos;
+        _logger._enable();
+        // Get the meta (for checking) and the object length (to know how much to read)
+        local marker = _flash.read(pos, SPIFLASHLOGGER_OBJECT_MARKER_SIZE).tostring();
+        local len = _flash.read(pos + SPIFLASHLOGGER_OBJECT_MARKER_SIZE, 2).readn('w');
+        _logger._disable();
+        _len = len;
+        _isValid = (marker == SPIFLASHLOGGER_OBJECT_MARKER && len > 0 && len < _logger._size);
+    }
+
+    function isValid() {
+        return _isValid;
+    }
+
+    function getAddress() {
+        return _addr;
+    }
+
+    function getPayload() {
+        if (!_isValid)
+            return null;
+
+        if (_payload != null)
+            return _payload;
+
+        local pos = _addr;
+        local serialised = blob(SPIFLASHLOGGER_OBJECT_HDR_SIZE + _len);
+        local leftInObject;
+        _logger._enable();
+        // while there is more object left, read as much as we can from each sector into `serialised`
+        while (leftInObject = serialised.len() - serialised.tell()) {
+            // Decide what to read
+            local sectorStart = pos - (pos % SPIFLASHLOGGER_SECTOR_SIZE);
+            local sectorEnd = sectorStart + SPIFLASHLOGGER_SECTOR_SIZE;
+            local leftInSector = sectorEnd - pos;
+
+            // Read it
+            local read;
+            if (leftInObject < leftInSector) {
+                read = _flash.read(pos, leftInObject);
+                assert(read.len() == leftInObject);
+            } else {
+                read = _flash.read(pos, leftInSector);
+                assert(read.len() == leftInSector);
+            }
+
+            serialised.writeblob(read);
+
+            // Update remaining and position
+            leftInObject -= read.len();
+
+            pos += read.len();
+            assert(pos <= sectorEnd);
+
+            if (pos == _logger._end) pos = _logger._start + SPIFLASHLOGGER_SECTOR_METADATA_SIZE;
+            else if (pos == sectorEnd) pos += SPIFLASHLOGGER_SECTOR_METADATA_SIZE;
+
+        }
+        _logger._disable();
+
+        // Try to deserialize the object
+        _payload = null;
+        try {
+            _payload = _logger._serializer.deserialize(serialised, SPIFLASHLOGGER_OBJECT_MARKER);
+        } catch (e) {
+            //server.error(format("Exception reading logger object address 0x%04x with length %d: %s", _addr, serialised.len(), e));
+            _payload = null;
```

Comment:
this is a redundant statement. if an exception was thrown _payload is not initialized.

Reply:
sounds reasonable, I'll drop payload = null;